src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -411,10 +411,14 @@ class AppsController(
 
   private def listVersions(appId: PathId)(implicit identity: Identity): Route = {
     val versions = groupManager.appVersions(appId).runWith(Sink.seq)
-    authorized(ViewRunSpec, groupManager.runSpec(appId).get, Rejections.EntityNotFound.noApp(appId)).apply {
-      onSuccess(versions) { versions =>
-        complete(VersionList(versions))
-      }
+    groupManager.app(appId) match {
+      case Some(app) =>
+        authorized(ViewRunSpec, app, Rejections.EntityNotFound.noApp(appId)).apply {
```

Comment:
@alenkacz  this is easier to read my opinion!

Question tho:   `case None` is entity not found... how do we have `case Some(app) with an `EntityNotFound`?  Should that be a not authorized?

Reply:
Good question, I don't know. I think it is like this because this is just a rewrite of old resources and I see this logic there as well. https://github.com/mesosphere/marathon/blob/e07df868b0b620e355ffdc498296c06a3f3b8a94/src/main/scala/mesosphere/marathon/api/v2/AppVersionsResource.scala#L45

maybe @meln1k will know more since he is author of this...

I personally would be for changing this to return not authorized here instead of not found to be consistent with the rest...

Reply:
I think that `NotFound` case will be covered by `PathMatcher`. So we probably should use not authorized in this case.